src/Controller/SantaController.php

```diff
@@ -54,90 +54,103 @@ public function __construct(RouterInterface $router, Environment $twig, LoggerIn
+            $availableUsers = $application->getUsers();
+
+            $session->set('available-users', $availableUsers);
+        }
+
+        $selectedUsers = $session->get('selected-users', []);
         $errors = [];
 
         if ($request->isMethod('POST')) {
             $selectedUsers = $request->request->get('users', []);
-            $message = $request->request->get('message');
 
-            $errors = $this->validate($selectedUsers, $message);
-
-            if (\count($errors) < 1) {
-                if ($request->request->has('notesRedirect')) {
-                    $request->getSession()->set('setup', [
-                        'selectedUsers' => $request->request->all('users'),
-                        'message' => $request->request->get('message'),
-                    ]);
-
-                    return $this->redirectToRoute('notes', [
-                        'application' => $application->getCode(),
-                    ]);
-                }
-
-                $associatedUsers = $rudolph->associateUsers($selectedUsers);
-                $hash = md5(serialize($associatedUsers));
-                $notes = $request->request->all('notes');
-
-                $secretSanta = new SecretSanta(
-                    $application->getCode(),
-                    $application->getOrganization(),
-                    $hash,
-                    array_filter($allUsers, function (User $user) use ($selectedUsers) {
-                        return \in_array($user->getIdentifier(), $selectedUsers, true);
-                    }),
-                    $associatedUsers,
-                    $application->getAdmin(),
-                    $message,
-                    $notes
-                );
-
-                $request->getSession()->set(
-                    $this->getSecretSantaSessionKey(
-                        $secretSanta->getHash()
-                    ), $secretSanta
-                );
-
-                return new RedirectResponse($this->router->generate('send_messages', ['hash' => $secretSanta->getHash()]));
+            if (\count($selectedUsers) > 1) {
+                $session->set('selected-users', $selectedUsers);
+
+                return $this->redirectToRoute('message', ['application' => $application->getCode()]);
             }
+
+            $errors[] = 'At least 2 users should be selected.';
         }
 
-        $content = $this->twig->render('santa/application/run_' . $application->getCode() . '.html.twig', [
+        $content = $this->twig->render('santa/application/participants_' . $application->getCode() . '.html.twig', [
             'application' => $application->getCode(),
-            'users' => $allUsers,
+            'users' => $availableUsers,
             'groups' => $application->getGroups(),
-            'admin' => $application->getAdmin(),
             'selectedUsers' => $selectedUsers,
-            'message' => $message,
             'errors' => $errors,
+            'step' => 1,
```

Comment:
I think it would be better to set this variable in the twig template itself, it doesn't add value here and it's only related with the breadcrumb.